cmd/ddev/cmd/debug-match-constraint_test.go

```diff
@@ -0,0 +1,28 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/ddev/ddev/pkg/exec"
+	"github.com/ddev/ddev/pkg/versionconstants"
+	"github.com/stretchr/testify/require"
+	"regexp"
+	"testing"
+)
+
+// TestDebugMatchConstraintCmd checks to see match-constraint behaves as expected
+// @see https://github.com/Masterminds/semver#checking-version-constraints
+func TestDebugMatchConstraintCmd(t *testing.T) {
+
+	versionRegex := regexp.MustCompile(`^v[0-9]+\.`)
+	if !versionRegex.MatchString(versionconstants.DdevVersion) {
+		t.Skip(fmt.Sprintf("Skipping because ddev version doesn't start with any valid version tag, it's '%v'", versionconstants.DdevVersion))
+	}
```

Comment:
`t.Skip` won't run because `versionconstants.DdevVersion = v0.0.0-overridden-by-make` when there is no tags.

Let me add another commit here. I'll add a better condition.

Reply:
that is a valid version, but it will fail the constraint I added, bug if you look at the previous failed ones, there version was the sha hash, not that one.

Reply:
See https://github.com/ddev/ddev/actions/runs/11617743434/job/32353619870#step:16:16831

```
=== RUN   TestDebugMatchConstraintCmd
    debug-match-constraint_test.go:16: 
        	Error Trace:	/home/runner/work/ddev/ddev/cmd/ddev/cmd/debug-match-constraint_test.go:16
        	Error:      	Received unexpected error:
        	            	exit status 1
        	Test:       	TestDebugMatchConstraintCmd
        	Messages:   	Match constraint should not have errored for = fc8df38, out='error: the '= fc8df38' constraint is not valid. See https://github.com/Masterminds/semver#checking-version-constraints for valid constraints format
        	            	'
--- FAIL: TestDebugMatchConstraintCmd (0.03s)
```

Reply:
`fc8df38` is the version

Reply:
So if with tags we get the proper version and without we get the hash, I think we are good.

Reply:
Your test failed here, but it shouldn't.

Reply:
> Your test failed here, but it shouldn't.

I mean, I ran your latest commit without tags, and it failed

```
=== RUN   TestDebugMatchConstraintCmd
    debug-match-constraint_test.go:27: 
        	Error Trace:	/home/stas/code/ddev/ddev-stasadev-build/cmd/ddev/cmd/debug-match-constraint_test.go:27
        	Error:      	An error is expected but got nil.
        	Test:       	TestDebugMatchConstraintCmd
        	Messages:   	Match constraint should have errored for < 1.0, out=''
--- FAIL: TestDebugMatchConstraintCmd (0.09s)
```

---

And now I replaced `versionconstants.DdevVersion` with `ddev --version`, and your regex works fine, and `t.Skip` works.

Reply:
there's a fix on another file unrelated to this PR, assume that's OK?

Reply:
Yes, I got it from my closed PR https://github.com/ddev/ddev/pull/6361
(I forgot about it before, I hope it's okay to add it here so I don't forget again 🙂)

Reply:
That's fine with me.. I think that with no tags, on the tests at least, you get the hash, I wonder why you got nil locally. If, however, I should not that if ever end up with `v0.0.0-overridden-by-make` the test will fail because of the test expectation that > 1.0 will always succeeded and also because of what you added before that:

Locally `GOTEST_SHORT=true make VERSION="v0.0.0-overridden-by-make" testcmd TESTARGS="-run TestDebugMatchConstraintCmd"`:

```
=== RUN   TestDebugMatchConstraintCmd
    debug-match-constraint_test.go:26: 
        	Error Trace:	/Users/ariel/Sites/work/ddev/cmd/ddev/cmd/debug-match-constraint_test.go:26
        	Error:      	Received unexpected error:
        	           	exit status 1
        	Test:       	TestDebugMatchConstraintCmd
        	Messages:   	Match constraint should not have errored for != v0.0.0-overridden-by-make, out='error: your DDEV version 'v0.0.0-overridden-by-make' doesn't meet the constraint '!= v0.0.0-overridden-by-make'. Please update to a DDEV version that meets this constraint 
```

Just saying

Reply:
> t.Skip won't run because versionconstants.DdevVersion = v0.0.0-overridden-by-make when there is no tags.

I said that, but I didn't check.

> I think that with no tags, on the tests at least, you get the hash

You are right. In reality `v0.0.0-overridden-by-make` is not set in any case.

We already have a check for `!= v0.0.0-overridden-by-make` here, in another test:

https://github.com/ddev/ddev/blob/00407fd09b9633a52001f94bc280c8d336bc3747/cmd/ddev/cmd/testdata/TestCmdAddonDdevVersionConstraint/valid_constraint_recipe/install.yaml#L3

And `TestCmdAddonDdevVersionConstraint` didn't fail on your PR with and without tags.

Actually, I think it's a good thing, because it ensures that this value is always changed to something else (except when you see it yourself with `make VERSION="v0.0.0-overridden-by-make"`).

https://github.com/ddev/ddev/blob/00407fd09b9633a52001f94bc280c8d336bc3747/pkg/versionconstants/versionconstants.go#L3-L4

So if this `TestDebugMatchConstraintCmd` or `TestCmdAddonDdevVersionConstraint` test fails, we'll immediately know that there's something wrong with make.

Thank you @hanoii for paying attention to this.